src/subdomains/core/statistic/cfp.service.ts

```diff
@@ -19,52 +17,66 @@ export interface CfpSettings {
   endDate: string;
 }
 
-interface CfpResponse {
-  number: number;
+export interface Proposal {
+  proposalId: string;
   title: string;
-  html_url: string;
-  labels: { name: string }[];
-  comments: number;
+  context: string;
+  contextHash: string;
+  status: ProposalStatus;
+  type: ProposalType;
+  amount: number;
+  payoutAddress: string;
+  currentCycle: number;
+  totalCycles: number;
+  cycleEndHeight: number;
+  proposalEndHeight: number;
+  votingPeriod: number;
+  quorum: string;
+  votesPossible: number;
+  votesPresent: number;
+  votesPresentPct: string;
+  votesYes: number;
+  votesYesPct: string;
+  approvalThreshold: string;
+  fee: number;
+  proposalVotes: Vote[];
 }
 
-interface CommentsResponse {
-  body: string;
-  created_at: string;
+export interface ProposalVote {
+  proposalId: string;
+  masternodeId: string;
+  cycle: number;
+  vote: VoteResult;
+}
+
+enum VoteResult {
+  YES = 'YES',
+  NO = 'NO',
+  NEUTRAL = 'NEUTRAL',
 }
 
 enum ResultStatus {
   APPROVED = 'Approved',
   NOT_APPROVED = 'Not approved',
 }
 
-enum State {
-  ENABLED = 'ENABLED',
-  PRE_ENABLED = 'PRE_ENABLED',
-}
-
 enum VotingType {
   CFP = 'cfp',
   DFIP = 'dfip',
 }
 
 interface Vote {
   address: string;
-  signature: string;
   cfpId: string;
   vote: string;
-  createdAt: string;
-  isCake: boolean;
-  isDfx: boolean;
-}
-
-export interface MasterNode {
-  ownerAuthAddress: string;
-  mintedBlocks: number;
-  state: State;
+  signature?: string;
+  createdAt?: string;
+  isCake?: boolean;
+  isLock?: boolean;
 }
 
 export interface CfpResult {
-  number: number;
+  number?: number;
```

Comment:
We should keep this identifier. Maybe use the proposal ID? UI's use this for voting.